_posts/2019-09-17-hhvm-4.23.0.markdown

```diff
@@ -0,0 +1,79 @@
+---
+title: "HHVM 4.23.0"
+layout: post
+author: fred
+category: blog
+---
+
+HHVM 4.23 is released! This release marks the end of support for 4.17;
+4.18-4.22 remain supported, as do the LTS releases 3.30 and 4.8.
+
+# Highlights
+
+- fixed runtime error calling static methods with reified generics on dynamic
+  class names, e.g. `$class::foo<int>()`
+- clearer typechecker error messages for invalid method and property access
+- clearer runtime error when calling an undefined method with `inout` parameters
+- if the admin server is enabled with a passsword, it is now possible to allow
+  access to statistics without the password with the
+  `hhvm.admin_server.stats_need_password` INI setting.
+
+# Breaking Changes
+
+- Fix variance checks for `where` constraints with generics
+- It is now a typechecker error to have multiple `default:` cases in a `switch`;
+  this was previously a runtime error
+- A typechecker error is now raised if a `default:` case is present in a
+  `switch` but is not the last case.
+- Deleted temporary APIs for reified generics:
+  - `HH\\ReifiedGenerics\\getTypeStructure()`: use
+    `HH\\ReifiedGenerics\\get_type_structure()` instead
+  - `HH\\ReifiedGenerics\\getClassname()`: use
+    `HH\\ReifiedGenerics\\get_classname()` instead
+- Accessing `$GLOBALS` without a subcript is now a parse error. Use
+  `HH\global_keys()` instead if you need to enumarate the full list.
+- The XML, memcache, memcached, and PCRE extensions have been migrated from
+  taking parameters by reference to taking `inout` parameters; in many cases
+  this is just a change to the calling convention, however there are
+  more significant changes, due to `inout` parameters not permitting default
+  values; in particular:
+  - `Memcache::get()` and `memcache_get()` no longer take a `&$flags` parameter;
+    this parameter previously had no effect in HHVM.
+  - `preg_replace()` no longer has an optional `$count` parameter;
+    call `preg_replace_with_count()` if a count is required.
```

Comment:
This one was already announced a few weeks ago, but I guess it doesn't hurt to mention it again...

Reply:
(`preg_replace()` hasn't had the `&$count` argument for 2--3 last versions already)